src/main/java/com/exasol/adapter/request/RefreshRequest.java

```diff
@@ -2,31 +2,34 @@
 
 import java.util.List;
 
+import com.exasol.adapter.AdapterException;
 import com.exasol.adapter.metadata.SchemaMetadataInfo;
 
 public class RefreshRequest extends AdapterRequest {
-    
+
     private boolean isRefreshForTables;
     private List<String> tables;
-    
+
     public RefreshRequest(SchemaMetadataInfo schemaMetadataInfo) {
         super(schemaMetadataInfo, AdapterRequestType.REFRESH);
         isRefreshForTables = false;
     }
-    
-    public RefreshRequest(SchemaMetadataInfo schemaMetadataInfo, List<String> tables) {
+
+    public RefreshRequest(SchemaMetadataInfo schemaMetadataInfo, List<String> tables) throws AdapterException {
         super(schemaMetadataInfo, AdapterRequestType.REFRESH);
-        assert(tables != null && !tables.isEmpty());
+        if (tables == null || tables.isEmpty()) {
+            throw new AdapterException("The list of tables is empty.");
```

Comment:
Please be more specific about the context of the error.
Imagine if you as a user read "The list of tables is empty."
You would like to know "Which table list? Where?"